src/libraries/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj

```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <TargetFrameworks>$(NetCoreAppCurrent);net48</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent)-Windows_NT;$(NetCoreAppCurrent);net48</TargetFrameworks>
```

Comment:
Seems unnecessary here since we don't copy the assembly, nor are we referencing the implementation.  Ideally we shouldn't need to do this.

Reply:
> Seems unnecessary here since we don't copy the assembly, nor are we referencing the implementation. Ideally we shouldn't need to do this.

We are referencing the implementation: https://github.com/dotnet/runtime/pull/42826/files/24598978763b798fcb8611251cc1afb834470078#diff-f65cfd9017749fb24c89920850f3e7e5R42.

> Why does this need to be Windows specific? Is it because we need to replicate the configurations of the src project?

Correct, otherwise we can't make this work with our current configuration system rules.

Reply:
I understand the mechanism you used was to ProjectReference to the implementation project, but the thing that is getting passed to the compiler is not that, it's the reference assembly.  I'm raising the fact that it's unusual that this is viral in this way.  Will people know they need to do this?  What happens in the case where they are actually building a test project that is netcoreappcurrent and a src project has netcoreappcurrent;netcoreappcurrent-windows?  What tells them they need to add a TFM in order to get the right behavior?

Reply:
> I'm raising the fact that it's unusual that this is viral in this way.

Do you have suggestions how to solve this in a better way? I'm not happy about this either as it means more project configurations and with that more builds "just" for the sake of a valid reference from the test to the source project.

Reply:
What if the tests use a different TFM for resolving the P2P in this case?  I believe it is the same case that you called out above.  When it's part of NetCoreApp, and you don't need to reference the implementation, then you're just rebuilding for convenience and you only care about rebuilding for the test's TargetOS.  Why not resolve the P2P as `$(NetCoreAppCurrent)-$(TargetOS)`?

Reply:
That would work but for inbox assemblies only and only P2Ps that aren't referencing the src directly (ie InternalsVisibleTo).

Reply:
I'm unsure if the configuration system respects the `SetTargetFramework` metadata?

Reply:
now it does. resolving. :D